runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -58,6 +59,9 @@ public static void subscribe(@Nonnull LogReplicationListener listener, @Nonnull
                                  int bufferSize, CorfuStore corfuStore) {
         log.info("Client subscription process has started.");
 
+        // Ensure status table is opened here in case FullSyncExecutor is late to do so.
+        openReplicationStatusTable(corfuStore);
+
```

Comment:
I'd prefer to modify the original `openReplicationStatusTable` helper method the be something like try getTable and if doesn't exist we can just open it, such that both the main thread and subscribe task could reuse it

Reply:
actually I'm a bit confused by the IRetry for openTable operation, I saw @hisundar 's comment, but the txn that open table will be doing is just writing to RegistryTable, if we check the registerTable method in TableRegistry, it already has its own retry logic. So @hisundar please correct me if I missed anything

so IMO the try - getTable - catch - openTable pattern you've been using in LogicalGroupClient should be fine for that helper method

Reply:
yes openTable does retry on TransactionAborts but it will also give up after certain retries and in those cases we should throw the `onError()` as @shreayp  is doing in latest patchset right?

Reply:
yes I see the point of `onError`, but doesn't that exactly mean we don't need the outer `IRetry` here? I didn't understand how could it help here if the `TxAbort` is already handled inside, and the `StreamingException` it is catching is not something could be fixed with retry right?